File: vdp-python/visa/test/vta/test_manage_notifications.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import unittest
import sys
import os
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestManageNotifications(unittest.TestCase):
   
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()
        self.notification_subscription_request = json.loads('''{
                         "contactType": "''' + self.config.get('VDP','vtaNotificationContactType') + '''",
                         "contactValue": "john@visa.com",
                         "emailFormat": "None",
                         "last4": "''' + self.config.get('VDP','vtaCreateCustomerLastFour') + '''",
                         "phoneCountryCode": "en-us",
                         "platform": "None",
                         "preferredLanguageCode": "''' + self.config.get('VDP','vtaPreferredLanguageCode') + '''",
                         "serviceOffering": "WelcomeMessage",
                         "serviceOfferingSubType": "WelcomeMessage",
                         "substitutions": {}
                      }''')
         
    def test_notification_subscription(self):
        base_uri = 'vta/'
        resource_path = 'v3/communities/'+ self.config.get('VDP','vtaCommunityCode') +'/portfolios/' + self.config.get('VDP','vtaPortfolioNumber') +'/customers/' + self.config.get('VDP','vtaCustomerId')+ '/notifications';
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.notification_subscription_request, 'Notification Subscriptions Test', 'post', {'ServiceId' : self.config.get('VDP','vtaServiceId')})
        self.assertEqual(str(response.status_code) ,"201" ,"Notification Subscriptions Test failed")
        pass

File: vdp-java/src/test/java/com/visa/vdp/vta/TestManageNotifications.java
package com.visa.vdp.vta;

import java.util.HashMap;
import java.util.Map;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestManageNotifications {

    String notificationSubscriptionRequest;
    VisaAPIClient visaAPIClient;

    @Test(groups = "vta")
    public void setUp() {
        this.visaAPIClient = new VisaAPIClient();
        this.notificationSubscriptionRequest = "{"
                        + "\"contactType\":  \""+ VisaProperties.getProperty(Property.VTA_CONTACT_TYPE)+ "\","
                        + "\"contactValue\":  \"john@visa.com\","
                        + "\"emailFormat\": \"None\","
                        + "\"last4\":  \""+ VisaProperties.getProperty(Property.VTA_CREATE_CUSTOMER_LAST_FOUR)+ "\","
                        + "\"phoneCountryCode\": \"en-us\","
                        + "\"platform\": \"None\","
                        + "\"preferredLanguageCode\":  \""+ VisaProperties.getProperty(Property.VTA_PREFFERED_LANGUAGE_CODE)+ "\","
                        + "\"serviceOffering\":  \"WelcomeMessage\","
                        + "\"serviceOfferingSubType\":  \"WelcomeMessage\","
                        + "\"substitutions\": {}"
                     + "}";
    }

    @Test(groups = "vta")
    public void testNotificationSubscription() throws Exception {
        String baseUri = "vta/";
        String resourcePath = "v3/communities/"+VisaProperties.getProperty(Property.VTA_COMMUNITY_CODE) +"/portfolios/" 
                + VisaProperties.getProperty(Property.VTA_PORTFOLIO_NUMER) +"/customers/" + VisaProperties.getProperty(Property.VTA_CUSTOMER_ID)
                + "/notifications";

        Map<String,String> headers = new HashMap<String,String>();
        headers.put("ServiceId", VisaProperties.getProperty(Property.VTA_SERVICE_ID));

        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Notification Subscriptions Test", this.notificationSubscriptionRequest, MethodTypes.POST, headers);
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED);
        response.close();
    }
}
